File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.kt
class Solution {
    fun solution(keymap: Array<String>, targets: Array<String>): IntArray {
        val answer = IntArray(targets.size) { -1 }

        for (i in targets.indices) {
            var count = 0

            for (j in targets[i]) {
                var index = 0

                if (keymap.count { it.contains(j) } == 0) {
                    count = -1
                    break
                }

                val list = keymap.filter { it.contains(j) }

                while (true) {
                    if (list.count { it[index] == j } > 0) {
                        count += index+1
                        break
                    }else {
                        index++
                    }
                }
            }
            answer[i] = count
        }
        return answer
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.swift
func solution(_ keymap:[String], _ targets:[String]) -> [Int] {
    var answer = [Int]()
    var dict = [Int: Set<String>]()
    
    for key in keymap {
        for (i, c) in key.enumerated() {
            if dict[i+1] != nil {
                dict[i+1]!.insert(String(c))
            } else {
                dict[i+1] = [String(c)]
            }
        }
    }
    
    for target in targets {
        var index = 0
        
        for c in target {
            let count = getTouchCount(String(c), dict)
            
            if count == -1 {
                index = -1
                break
            } else {
                index += count
            }
        }
        
        answer.append(index)
    }
    
    return answer
}

func getTouchCount(_ s: String, _ dict: [Int: Set<String>]) -> Int {
    for i in dict.keys.sorted() {
        if dict[i]!.contains(s) {
            return i
        }
    }
    
    return -1
}